src/Objects/User.php

```diff
@@ -14,6 +20,57 @@
 class User extends BaseObject
 {
     /**
+     * @var string $currentConversation
+     */
+    private $currentConversation;
+
+    /**
+     * @var SQLite3Cache $cache
+     */
+    private $cache;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct($data)
+    {
+        parent::__construct($data);
+
+        $this->cache = new FilesystemCache(__DIR__.'/../Storage/cache');
```

Comment:
I'm not quite sure if it's any better because I haven't really used `doctrine/cache` directly. I've only used `illuminate/cache` in Laravel projects.

The problem with `illuminate` components is basically when the project is used within some incompatible version of Laravel, then we're forced to update the changes to our project as well as maintain multiple versions just to support Laravel users, which is kinda problem.

If `doctrine/cache` can support multiple drivers and it's easier, then I don't have any issues going with it.
The default could be file cache while giving users a choice to switch to Redis / Memcached / DB (SQLite or MySQL if that's possible).

Can we detect the platform and maybe use Laravel's built-in cache and settings when it's being used with Laravel and fallback to `doctrine/cache` for standalone users? That would make things a lot easier.